packages/mui-base/src/Accordion/Root/AccordionRoot.tsx

```diff
@@ -0,0 +1,185 @@
+'use client';
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { BaseUIComponentProps } from '../../utils/types';
+import { useComponentRenderer } from '../../utils/useComponentRenderer';
+import { CompositeList } from '../../Composite/List/CompositeList';
+import { useAccordionRoot } from './useAccordionRoot';
+import { AccordionRootContext } from './AccordionRootContext';
+
+/**
+ *
+ * Demos:
+ *
+ * - [Accordion](https://base-ui.netlify.app/components/react-accordion/)
+ *
+ * API:
+ *
+ * - [AccordionRoot API](https://base-ui.netlify.app/components/react-accordion/#api-reference-AccordionRoot)
+ */
+const AccordionRoot = React.forwardRef(function AccordionRoot(
+  props: AccordionRoot.Props,
+  forwardedRef: React.ForwardedRef<HTMLDivElement>,
+) {
+  const {
+    animated,
+    className,
+    direction,
+    disabled = false,
+    hiddenUntilFound = false,
+    loop,
+    onValueChange,
+    openMultiple = true,
+    orientation,
+    value,
+    defaultValue: defaultValueProp,
+    render,
+    ...otherProps
+  } = props;
+
+  // memoized to allow omitting both defaultValue and value
+  // which would otherwise trigger a warning in useControlled
+  const defaultValue = React.useMemo(() => {
+    if (value === undefined) {
+      return defaultValueProp ?? [];
+    }
+
+    return undefined;
+  }, [value, defaultValueProp]);
+
+  const { getRootProps, ...accordion } = useAccordionRoot({
+    animated,
+    direction,
+    disabled,
+    defaultValue,
+    loop,
+    orientation,
+    onValueChange,
+    openMultiple,
+    value,
+  });
+
+  const ownerState: AccordionRoot.OwnerState = React.useMemo(
+    () => ({
+      value: accordion.value,
+      disabled: accordion.disabled,
+      orientation: accordion.orientation,
+    }),
+    [accordion.value, accordion.disabled, accordion.orientation],
+  );
+
+  const contextValue: AccordionRootContext = React.useMemo(
+    () => ({
+      ...accordion,
+      hiddenUntilFound,
+      ownerState,
+    }),
+    [accordion, hiddenUntilFound, ownerState],
+  );
+
+  const { renderElement } = useComponentRenderer({
+    propGetter: getRootProps,
+    render: render ?? 'div',
+    className,
+    ownerState,
+    ref: forwardedRef,
+    extraProps: otherProps,
+    customStyleHookMapping: {
+      disabled: (isDisabled) => {
+        if (isDisabled) {
+          return { 'data-disabled': '' };
+        }
+        return null;
+      },
+      value: () => null,
+    },
```

Comment:
Should be placed outside the component. I don't think `data-disabled` needs a custom style hook as this should work by default?

Reply:
@atomiks Moved - by default `disabled` becomes `data-disabled="true"` instead of just `data-disabled` which seems a bit better, maybe the `getStyleHookProps` util could be updated to handle this universally?

Reply:
I'm ok with just `data-disabled` and whatever we choose, we should apply to all components. So `getStyleHookProps` should handle it.

Reply:
https://github.com/mui/base-ui/pull/577/commits/2c8327eea72f75a757ad7230f21175a43e119219
Since we already don't render any data attr when the value is `false`, I think we could change all `data-attr='true'` to just `data-attr` (e.g. `data-readonly`, `data-required`...)